### PR TITLE
GitHub dependency removement, add ing GitLab Bitbucket detection

### PR DESCRIPTION
**Pull request by voatsap**
- removed GITHUB_* dependencies in code
- implemented `detect_git_provider` function which sets the required variables in Github GitLab Bitbucket hosting providers

- @romanprog should redefine /tests/ regarding these changes

### CODE REVIEW

File: bin/common.sh [hunk 1]
```diff
@@ -24,3 +24,60 @@ function output_software_info {
...
+            readonly GIT_REPO_NAME=$(echo $BITBUCKET_GIT_HTTP_ORIGIN | sed -e 's/http:\/\/bitbucket.org\///g')
+            readonly GIT_REPO_ROOT=$BITBUCKET_CLONE_DIR
+    else
+        DEBUG "BITBUCKET_GIT_HTTP_ORIGIN variable is NOT set assuming we are NOT running Bitbucket Pipeline";
+fi
+
+# Output final results with required variables set
+INFO "GIT_PROVIDER is set for: $GIT_PROVIDER"
+INFO "GIT_REPO_NAME is set for: $GIT_REPO_NAME"
+INFO "GIT_REPO_ROOT is set for: $GIT_REPO_ROOT"
```
Comment (MaxymVlasov) [inline bin/common.sh:82]: Where `GIT_REPO_ROOT` used?

File: entrypoint.sh [hunk 2]
```diff
@@ -10,19 +10,19 @@ source "$PRJ_ROOT"/bin/aws_minikube.sh
...
 
-# Variables passed by Github Workflow to Action
+# Mandatory variables passed to container by config
 readonly CLUSTER_CONFIG_PATH=$1
 readonly CLOUD_USER=$2
 readonly CLOUD_PASS=$3
-readonly GIT_REPO_ROOT=$GITHUB_WORKSPACE
-# TODO: Create function to select git repo root from ENV for GitLab and BitBucket
 
+# Detect Git hosting and set GIT_REPO_NAME, GIT_REPO_ROOT variables
```
Comment (MaxymVlasov) [inline entrypoint.sh:18]:
```suggestion
# Detect Git hosting and set GIT_PROVIDER, GIT_REPO_NAME, GIT_REPO_ROOT constants
```

Comment (voatsap) [reply entrypoint.sh:18]: > 

At least at  ArgoCD  argocd::deploy_apps, or you think we could ovveride this with relative path?

File: bin/common.sh (hunk 1 above)
Comment (MaxymVlasov) [reply bin/common.sh:82]: voatsap 23 minutes ago Author Member

At least at ArgoCD argocd::deploy_apps, or you think we could ovveride this with relative path?

**Pull request merged**